addons/hr_holidays_attendance/models/hr_leave_allocation.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from odoo.tools import float_round
+
+
+class HolidaysAllocation(models.Model):
+    _inherit = 'hr.leave.allocation'
+
+    overtime_deductible = fields.Boolean(compute='_compute_overtime_deductible')
+    overtime_id = fields.Many2one('hr.attendance.overtime', string='Extra Hours', groups='hr_holidays.group_hr_holidays_user')
+    employee_overtime = fields.Float(related='employee_id.total_overtime')
+    hr_attendance_overtime = fields.Boolean(related='employee_id.hr_attendance_overtime')
+
+    @api.depends('holiday_status_id')
+    def _compute_overtime_deductible(self):
+        deductibles = self.filtered(lambda a: a.holiday_status_id.company_id.hr_attendance_overtime and a.holiday_status_id.overtime_deductible)
+        deductibles.overtime_deductible = True
+        (self - deductibles).overtime_deductible = False
+
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        if 'holiday_status_id' in fields and self.env.context.get('deduct_extra_hours'):
+            leave_type = self.env['hr.leave.type'].search([('valid', '=', True), ('overtime_deductible', '=', True), ('allocation_type', '!=', 'no')], limit=1)
+            res['holiday_status_id'] = leave_type.id
+        return res
+
+    @api.model
+    def create(self, vals):
```

Comment:
Could you try to make this work in batch please?

Reply:
Parent model also have api.model on create, I will check if we can replace it with a multicreate